src/vs/workbench/api/common/shared/webview.ts

```diff
@@ -45,6 +45,11 @@ export function asWebviewUri(
 	remoteInfo?: { authority: string | undefined, isRemote: boolean }
 ): vscode.Uri {
 	if (resource.scheme === Schemas.http || resource.scheme === Schemas.https) {
+		const token = RemoteAuthorities.getConnectionToken(resource.authority);
+		if (token && !resource.query) {
```

Comment:
Adding this token to every http url seems like it could be problematic. Is there a way to restrict it to just the known server address?

Also, does this only happen when we are loading files from VS Code itself? Do images from extensions also have this issue?

Reply:
It's by authority, so it should only be the authorities which have registered a token with `RemoteAuthoritiesImpl#setConnectionToken(authority: string, connectionToken: string)`. I'm not sure about extensions